liiweb/views/legislation.py

```diff
@@ -8,25 +8,43 @@
 
 from peachjam.helpers import chunks, get_language
 from peachjam.models import JurisdictionProfile, Legislation, Locality, pj_settings
-from peachjam_api.serializers import LegislationSerializer
+from peachjam.views import FilteredDocumentListView
 
 
-class LegislationListView(TemplateView):
+class LegislationListView(FilteredDocumentListView):
     template_name = "liiweb/legislation_list.html"
     variant = "current"
     navbar_link = "legislation"
     model = Legislation
+    queryset = Legislation.objects.prefetch_related(
+        "taxonomies", "taxonomies__topic", "work"
+    )
     extra_context = {"doc_table_citations": True, "legislation_list_sort": "title"}
 
     def get_queryset(self):
+        qs = super().get_queryset()
         qs = (
-            self.model.objects.exclude(published=False)
-            .distinct("work_frbr_uri")
-            .order_by("work_frbr_uri", "-date", "language__pk")
+            qs.exclude(published=False)
+            .distinct(
+                "title",
+                "work_frbr_uri",
+            )
+            .order_by(
+                "title",
+                "work_frbr_uri",
+                "-date",
+                "language__pk",
+            )
             .preferred_language(get_language(self.request))
         )
+
         return qs
 
+    def get_template_names(self):
+        if self.request.htmx:
+            return ["peachjam/_document_table.html"]
```

Comment:
the view code needs to be able to differentiate between a refresh of the facets + table, or just a refresh of the table